app/uk/gov/hmrc/gform/controllers/helpers/FormDataHelpers.scala

```diff
@@ -59,8 +62,9 @@ object FormDataHelpers {
           }))
       }) match {
       case Some(requestData) =>
-        val (variadicFormData, requestRelatedData) = buildVariadicFormDataFromBrowserPostData(formModel, requestData)
-        continuation(requestRelatedData)(variadicFormData)
+        val (variadicFormData, requestRelatedData) =
+          buildVariadicFormDataFromBrowserPostData(formModelRenderPageOptics.formModel, requestData)
+        continuation(requestRelatedData)(formModelRenderPageOptics.recData.variadicFormData ++ variadicFormData)
```

Comment:
All the logic happens on this line, rest is just a removing things or white space difference